code/modules/mob/living/living.dm

```diff
@@ -443,6 +443,12 @@
 	if (s_active && !(s_active.ClickAccessible(src, depth=STORAGE_VIEW_DEPTH) || s_active.Adjacent(src)))
 		s_active.close(src)
 
+	for(var/mob/M in oview(src))
+		if(M.client && (src in cone(M, OPPOSITE_DIR(M.dir), view(10, M))))
```

Comment:
kor this isn't what I meant, you don't need to update the entire vision cone for a mob who isn't moving - they just need to know that this mob moved and then they need to check if the currently moving mob has entered or exited their cone zone and if so they need to hide or unhide appropriately